cibuildwheel/docker_container.py

```diff
@@ -34,7 +49,13 @@ class DockerContainer:
     bash_stdout: IO[bytes]
 
     def __init__(
-        self, *, docker_image: str, simulate_32_bit: bool = False, cwd: Optional[PathOrStr] = None
+        self,
+        *,
+        docker_image: str,
+        simulate_32_bit: bool = False,
+        cwd: Optional[PathOrStr] = None,
+        container_engine: str = "docker",
+        env: Optional[Dict[str, str]] = None,
```

Comment:
I think it is a logical extension of the object that may prove useful in the future. I feel like an object that is easy to test without framework specific test-time dependencies has advantages. I would be in inclined to leave this as-is, I try to avoid using pytest features when possible, but I'm not going to die on that hill.

Reply:
My perspective is more along the lines of YAGNI (you ain't gonna need it), and we use setenv quite a lot already. I prefer setenv because it's closer to how the code would perform if the user set the environment before calling cibuildwheel.

Reply:
I removed the `.env` attribute and modified the tests to use `monkeypatch.setenv`. Personally, I think the previous method was a lot cleaner, and the monkeypatch fixture adds unnecessary obfuscation (there is no in-code indication of where it comes from!), but I understand the value of limiting the complexity of the main class, so I'll compromise.

Reply:
thanks! I've refactored the monkeypatch.setenv calls a bit, I think they're a bit neater now.